src/error.rs

```diff
@@ -49,3 +49,11 @@ pub enum RoutingError {
     #[error(display = "Error while trying to receive a message from a mpsc channel.")]
     MpscRecvError(mpsc::RecvError),
 }
+
+// TODO dirvine, we need to complete the error story in our code. Here I am unboxing and that will
```

Comment:
This part is one I would like @jeanphilippeD  and @madadam  to look at a bit closer if possible. It feels a bit dirty, but let's see.

Reply:
Ok replacing the Error with `Box<ErrorKind>` is sufficient to remove that code.